Gems/Atom/Tools/MaterialCanvas/Assets/MaterialCanvas/GraphData/Nodes/MaterialOutputs/BasePBR/HandleCastShadows.lua

```diff
@@ -14,11 +14,13 @@ function GetMaterialPropertyDependencies()
 end
 
 function Process(context)
-    -- The only reason we need this script is when the material type uses GeneralCommonPropertyGroup.json
-    -- which does not connect general.cast_shadows, and the only way to add a connection after importing
-    -- a common .json file is to use a lua functor. If we had another way to add a connection to a property
-    -- after defining it via $import, that would probably be simpler.
-    local castShadows = context:GetMaterialPropertyValue_bool("general.cast_shadows")
+    local castShadows = true
+    if context:HasMaterialProperty("general.cast_shadows") then
+        castShadows = context:GetMaterialPropertyValue_bool("general.cast_shadows")
+    end
+
     local shadowMap = context:GetShaderByTag("Shadowmap")
-    shadowMap:SetEnabled(castShadows)
+    if shadowMap then
```

Comment:
should this if check castShadows as well before enabling the shadowmap?

Reply:
No because it's enabling or disabling it based on the flag